Mesh_3/include/CGAL/Mesh_3/polylines_to_protect.h

```diff
@@ -335,8 +363,8 @@ namespace CGAL {
   CGAL_assertion(nb_facets == expected_nb_facets);
   CGAL_USE(nb_facets); CGAL_USE(expected_nb_facets);
 
-
-  internal::Mesh_3::split_in_polylines(graph, polylines, K());
+  CGAL::Mesh_3::Polyline_visitor<Point_3, Graph> visitor(polylines, graph);
+  CGAL::split_graph_into_polylines(graph, visitor);
```

Comment:
The best solution to that the would be C++11 and `std::move`. If Boost Graph support the move constructors.



Le 26 mai 2016 16:25, à 16:25, Sebastien Loriot <notifications@github.com> a écrit:
>> @@ -335,8 +363,8 @@ namespace CGAL {
>>    CGAL_assertion(nb_facets == expected_nb_facets);
>>    CGAL_USE(nb_facets); CGAL_USE(expected_nb_facets);
>>  
>> -
>> -  internal::Mesh_3::split_in_polylines(graph, polylines, K());
>> +  CGAL::Mesh_3::Polyline_visitor<Point_3, Graph> visitor(polylines,
>graph);
>> +  CGAL::split_graph_into_polylines(graph, visitor);
>
>@lrineau Actually the graph is also copied internally in
>`split_graph_into_polylines` while the former implementation was doing
>the modification in-place. If that matters, another version of
>`split_graph_into_polylines` working in-place should be added.
>
>---
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/CGAL/cgal/pull/1110/files/897294f1c64291563f388e3192d24732f339f529#r64754574